scripts/needs_testrun.py

```diff
@@ -35,9 +33,8 @@ def get_base_branch(pr_number: int) -> str:
     '1.x'
     """
 
-    pr_page_content = urlopen(f"https://github.com/DataDog/dd-trace-py/pull/{pr_number}").read().decode("utf-8")
-
-    return BASE_BRANCH_PATTERN.search(pr_page_content).group(1)
+    response = json.load(urlopen(f"https://api.github.com/repos/datadog/dd-trace-py/pulls/{pr_number}"))
```

Comment:
This is neat but would consume some of the free quota we are allowed, causing the more important API call for the file list to potentially fail. I guess we can experiment with this and see what happens 🤞

Reply:
Good point! Will keep an eye out for exceptions.